aten/src/ATen/core/dispatch/OpSchema.h

```diff
@@ -88,6 +115,50 @@ struct has_name_defined<T, guts::void_t<
 
 // TODO Test has_name_defined
 
+template<class ReturnType, class ParamTypes, class FuncType> struct _wrapKernel {};
+template<class ReturnType, class... ParamTypes, class FuncType> struct _wrapKernel<ReturnType, guts::typelist::typelist<ParamTypes...>, FuncType> {
+  using parameter_types = guts::typelist::typelist<ParamTypes...>;
+
+  template<size_t... indices>
+  static KernelFunction call(FuncType* kernel, guts::index_sequence<indices...>) {
+    return [kernel] (ArrayRef<IValue> args) -> IValue {
+      if (args.size() != sizeof...(ParamTypes)) {
+        throw std::runtime_error("Wrong number of arguments for operator call");
+      }
+      return return_type_to_ivalue(
+        (*kernel)(args[indices].to<guts::remove_cv_t<guts::remove_reference_t<guts::typelist::element_t<indices, parameter_types>>>>()...)
+      );
+    };
+  }
+};
+template<class... ParamTypes, class FuncType> struct _wrapKernel<void, guts::typelist::typelist<ParamTypes...>, FuncType> {
```

Comment:
Once you actually make it take a Stack, you won't need two overloads, right?

Reply:
that's correct